src/DGtal/shapes/Mesh.ih

```diff
@@ -435,14 +437,62 @@ DGtal::Mesh<TPoint>::clearFaces(){
   myFaceList.clear();
 }
 
+
 template<typename TPoint>
 inline
 void
-DGtal::Mesh<TPoint>::changeScale(const typename TPoint::Component aScale){
-  for(typename VertexStorage::iterator it = vertexBegin(); it != vertexEnd(); it++)
+DGtal::Mesh<TPoint>::clearVertices(){
+  myVertexList.clear();
+}
+
+template <typename TPoint>
+void
+DGtal::Mesh<TPoint>::removeIsolatedVertices(){
+    typedef typename Mesh<TPoint>::Index MIndex;
+    DGtal::Mesh<TPoint>::VertexStorage vSt;
+    std::vector<bool> vertexUsed (nbVertex(), false);
+    for ( MIndex f = 0; f< nbFaces(); f++ )
     {
-      (*it) *= aScale;
+        auto face = getFace(f);
+        for (MIndex i = 0; i<face.size(); i++)
+        {
+            vertexUsed[face[i]] = true;
+        }
+    }
+    std::vector<int> translateIndexId;
+    MIndex currentIndex = 0;
+    MIndex nbV = nbVertex();
+    for(MIndex i = 0; i < nbV; i++ )
+    {
+        if (vertexUsed[i])
+        {
+            translateIndexId.push_back(currentIndex);
+            currentIndex++;
+        }
+        else
+        {
+            translateIndexId.push_back(0);
+            myVertexList.erase(myVertexList.begin()+currentIndex);
+        }
     }
+    for ( MIndex f = 0; f< nbFaces(); f++ )
+    {
+        auto &face = getFace(f);
+        for (MIndex i = 0; i<face.size(); i++)
+        {
+            face[i]=translateIndexId[face[i]];
+        }
+    }
+}
+
+template<typename TPoint>
+inline
+void
+DGtal::Mesh<TPoint>::changeScale(const typename TPoint::Component aScale){
```

Comment:
The method name is not great... `rescale` ?

Reply:
sure ;)